subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/LauncherServices.java

```diff
@@ -89,6 +91,10 @@ ExecuteBuildActionRunner createExecuteBuildActionRunner() {
         ClassLoaderCache createClassLoaderCache() {
             return new ClassLoaderCache();
         }
+
+        ScheduledExecutorService createScheduledExecutorService() {
+            return Executors.newScheduledThreadPool(1);
```

Comment:
There are already several other things that use this executor service (e.g. daemon expiration checks, GC stats polling).  Instead of setting the number of threads to a value that may start aging immediately, I've set the keep alive to 10s.  This should ensure that anything that runs at a frequency lower than that will always get a warm thread and we won't be constantly stopping/starting threads.